test/jdk/java/awt/TrayIcon/TrayIconScalingTest.java

```diff
@@ -0,0 +1,126 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8255439
+ * @key headful
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary To test tray icon scaling with on-the-fly DPI/Scale changes on Windows
+ * @run main/manual TrayIconScalingTest
+ * @requires (os.family == "windows")
+ */
+
+import java.awt.font.TextLayout;
+import java.awt.image.BaseMultiResolutionImage;
+import java.awt.image.BufferedImage;
+import java.awt.*;
+import java.awt.image.MultiResolutionImage;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+
+public class TrayIconScalingTest {
+
+    private static final String INSTRUCTIONS =
+            "This test case checks the scaling of tray icons for on-the-fly" +
+                    " DPI/ Scale changes on Windows.\n\n" +
+                    "STEPS: \n\n" +
+                    "1. When you run this test check the system tray/" +
+                    " notification area on windows, a white multi-resolution" +
+                    " image (MRI) icon should be visible.\n\n"+
+                    "2. Navigate to Settings > System > Display and change the" +
+                    " display scale by selecting any value from" +
+                    " Scale & Layout dropdown.\n\n"+
+                    "3. On scale changes observe the white tray icon," +
+                    " if there is NO distortion then press PASS.\n\n";
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        PassFailJFrame passFailJFrame = new PassFailJFrame("TrayIcon " +
+                "Test Instructions", INSTRUCTIONS, 8, 18, 85);
+        createAndShowGUI();
+        passFailJFrame.awaitAndCheck();
+
+    }
+
+    private static void createAndShowGUI() {
+
+        // Check if SystemTray supported on the machine
+        if (!SystemTray.isSupported()) {
+            System.out.println("SystemTray is not supported");
+            return;
+        }
+
+        // Create Multi Resolution Image
+        ArrayList<Image> images = new ArrayList<>();
+        for (int size = 16; size <= 34; size++) {
+            createIcon(size, images);
+        }
+        MultiResolutionImage multiResolutionImage =
+                new BaseMultiResolutionImage(images.toArray(new Image[0]));
+
+        SystemTray tray = SystemTray.getSystemTray();
+        TrayIcon icon = new TrayIcon((Image) multiResolutionImage);
+
+        PopupMenu popup = new PopupMenu();
+        MenuItem exitItem = new MenuItem("Exit");
+        popup.add(exitItem);
+        icon.setPopupMenu(popup);
+
+        try {
+            tray.add(icon);
+        } catch (AWTException e) {
+            throw new RuntimeException("Error while adding icon to system tray");
+        }
+
+        exitItem.addActionListener(e -> {
+            tray.remove(icon);
+            System.exit(0);
```

Comment:
Even if it is manual etc, we don't call System.exit() in jtreg tests .. we just shut down all the windows and dispose them so there is nothing to keep the main thread from exiting.
If this is called, I don't know why Kevin didn't see the app exit.
I suspect it cannot have been and he exited via some other path which left windows in existence which meant the toolkit thread didn't exit.

Reply:
The `actionListener` is only called if you right click on the tray icon and select the Exit option. I didn't do that, so it never got here. Maybe the "Exit" button in the icon should be turned into a no-op, since you want the tester to press Pass or Fail to exit?

I think Alexey may have pointed out the problem with the test not exiting: it doesn't dispose the tray icon. That needs to be confirmed.